### PR TITLE
feat: lu file support reference

### LINKED ISSUE #3668: Composer should support lu file references
## Is your feature request related to a problem? Please describe.

The [.lu format](https://docs.microsoft.com/en-us/azure/bot-service/file-format/bot-builder-lu-file-format?view=azure-bot-service-4.0) used in composer support "exteneral references", which could be helpful in certain cases where user want to share there .lu content to other .lu files. But Composer don't support that, either referencing another .lu file or a certain section of a .lu file will result in error in Composer.
 
## Describe the solution you'd like

Support "external references"  in .lu files.  By implementing the resource resolver the lu parser needed to resolve those external references.

## Additional context
There might be additional work to support different kinds references, ie, whole file reference or section reference. In that case, whole file reference has a higher priority since it's the syntax that expected to be used in most cases.

### PR DESCRIPTION
**Pull request by zhixzhan**
## Description

Composer implement support refer lu file by id, and locale can be ignored.

##### refer by wildcard matcher \*/\**
- [x]  `[link name](<path>/*)` 
- [x]  `[link name](<path>/**)` 

##### refer a file
- [x]  `[link name](help.lu)` refer by id
- [x]  `[link name](help.en-us.lu)` when not specify locale will auto fallback to current luFile's locale 
- [x]  `[link name](./help.lu)` with path

##### refer a intent's utterances and patterns
- [x] `[link name](help.lu#HowToDrive)`
- [x] `[link name](help.lu#HowToDrive*utterances*)`
- [x] `[link name](help.lu#HowToDrive*patterns*)`

##### refer file's utterances and patterns
- [x] `[link name](help.lu#*utterances*)`
- [x] `[link name](help.lu#*patterns*)`
- [x] `[link name](help.lu#*utterancesAndPatterns*)`

### QnA reference
- [ ] `[link name](help.qna#$name?)`
- [ ] `[link name](help.qna#*alterations*?)`
- [ ] `[link name](help.qna#?question-to-find?)`
- [ ] `[link name](help.qna#*answers*?)`

## Known issue
1. Due to LU parser not support customized importResolver yet, client / LUEditor are not be able do intelligence auto-complete or find out imported file not existed. e.g. 
```
[link name](helpNotExist.lu)
```
only when publishing time  check error. this is tracked via https://github.com/microsoft/botframework-cli/issues/933

2. if a referenced intent not exist, the referenced' utterances will be empty,  and do not block publishing. e.g.  
```
[link name](help.lu#NotExistIntent)
```

## Task Item

close #3668

### CODE REVIEW

**Comment by hibrenda** [comment]
Closing this https://github.com/microsoft/BotFramework-Composer/issues/2241 as well

File: Composer/packages/server/src/models/bot/luResolver.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+   *   [help](help.lu)
+   *
+   *  would resolve to help.en-us.lu || help.lu
+   *
+   *  sourceId =  todosample.en-us.lu
+   *  idsToFind =   [{ filePath: help.lu, includeInCollate: true}, ...]
+   *
+   *  Overlap implemented built-in fs resolver in
+   *  botframework-cli/packages/lu/src/parser/lu/luMerger.js#findLuFilesInDir
+   *  Diffrence is composer support import by id, but not support * / ** file match
```
[the remark this answers is not in the document]
Comment (zhixzhan) [reply Composer/packages/server/src/models/bot/luResolver.ts:47]: Did you mean also support wildcard match in Composer?  A dialog's lu reference may looks like
```
./*           import current folder files
./**          import current folder & sub folder files
../**         import current dialog all locale files
../fr-fr/*    import current dialog some locale files       
../../language-understanding/en-us/*
../../../dialogA/language-understanding/**
```
It requires user understand our bot file structure
```
    /[dialog]
        [dialog].dialog
        /dialogs/[childs]
        /language-generation
            /[locale]
                 [dialog].[locale].lg
        /language-understanding
            /[locale]
                 [dialog].[locale].lu
```
I thought our import resolve was file id based like lg, if we need support wildcard match */**, I can complete it.

[the remark this answers is not in the document]
Comment (zhixzhan) [reply Composer/packages/server/src/models/bot/luResolver.ts]: supported <path>* / **

**Comment by zhixzhan** [comment]
This PR's changes has been include in #3919 , shall we close this and deliver together with `QnA import url` feature PR ?

**Pull request closed without merging**